tilgangskontroll-tiltaksarrangor/src/main/kotlin/no/nav/amt/tiltak/tilgangskontroll_tiltaksarrangor/tilgang/metrics/AnsattRolleMetricRepository.kt

```diff
@@ -0,0 +1,45 @@
+package no.nav.amt.tiltak.tilgangskontroll_tiltaksarrangor.tilgang.metrics
+
+import no.nav.amt.tiltak.core.domain.tilgangskontroll.ArrangorAnsattRolle
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+
+@Repository
+open class AnsattRolleMetricRepository(
+	private val template: NamedParameterJdbcTemplate
+) {
+
+	fun antallAnsatteSomErKunKoordinator() = tellAnsatteMedRoller(
+			rolleSomSkalTelles = ArrangorAnsattRolle.KOORDINATOR,
+			rolleSomIkkeSkalTelles = ArrangorAnsattRolle.VEILEDER
+		)
+
+	fun antallAnsatteSomErKunVeileder() = tellAnsatteMedRoller(
+			rolleSomSkalTelles = ArrangorAnsattRolle.VEILEDER,
+			rolleSomIkkeSkalTelles = ArrangorAnsattRolle.KOORDINATOR
+		)
+
+	fun antallAnsatteSomHarBeggeRollene() = tellAnsatteMedRoller(
+			rolleSomSkalTelles = ArrangorAnsattRolle.VEILEDER,
+			rolleSomIkkeSkalTelles = ArrangorAnsattRolle.KOORDINATOR,
+			tellDeSomHarBegge = true,
+		)
+
+	private fun tellAnsatteMedRoller(
+		rolleSomSkalTelles: ArrangorAnsattRolle,
+		rolleSomIkkeSkalTelles: ArrangorAnsattRolle,
+		tellDeSomHarBegge: Boolean = false
+	): Int {
+		val not = if (tellDeSomHarBegge) "" else "not"
+
+		val sql = """
+			select count(distinct ansatt_id) as antall
```

Comment:
Hva med å putte alt i en spørring? Det er en sammensatt metrikk og tenker det er fint å kjøre bare en spørring mot dabasen som er skreddersydd for formålet med metrikken som du skal lage

Reply:
Kan se på det, men jeg er ikke så stor fan av de sammensatte spørringene for metrikkene.